src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -63,13 +60,13 @@ public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd){
                     return expenseOB.fail(expenseODFailEdit);}}
 
             if (expenseIDAdd.getIsRecurringExpense()) {
-                Expense newrecurExpense = expenseFactory.createMonthObject(setExpenseCreatorInfo(expenseIDAdd));
+                Expense newrecurExpense = (Expense) expenseFactory.createMonthObject(setExpenseCreatorInfo(expenseIDAdd));
                 session.addRecurExpense(newrecurExpense);
                 month.addExpense(newrecurExpense);
                 ExpenseOD expenseODSuccessAdd = new ExpenseOD("You have created a new recurring expense!");
                 return expenseOB.success(expenseODSuccessAdd);
             }else{
-                month.addExpense(expenseFactory.createMonthObject(setExpenseCreatorInfo(expenseIDAdd)));
+                month.addExpense((Expense) expenseFactory.createMonthObject(setExpenseCreatorInfo(expenseIDAdd)));
```

Comment:
I think every time you call the factory you should make a new variable and cast it to Expense and then add it to the month (like you did with newRecurExpense. I think this would make the code more readable than three layered method calls.